pg_auth_mon.c

```diff
@@ -420,8 +482,10 @@ _PG_init(void)
 void
 _PG_fini(void)
 {
+	log_pg_auth_mon_data();
+
```

Comment:
If we take pg_stat_statements as an example than we should not dump stats from _PG_fini():
https://github.com/postgres/postgres/blob/16bd4becee32240d09db5c6cbec87957fdfcd2d9/contrib/pg_stat_statements/pg_stat_statements.c#L474-L486

```suggestion
```